src/main/java/com/salesforce/dataloader/action/visitor/BulkApiVisitorUtil.java

```diff
@@ -111,6 +138,19 @@ void createJob(Config cfg) throws AsyncApiException {
             if (assRule != null && (assRule.length() == 15 || assRule.length() == 18)) {
                 job.setAssignmentRuleId(assRule);
             }
+        } else if (op == OperationEnum.query || op == OperationEnum.queryAll) {
+            if (this.enablePKChunking) {
+                String startRowParam = "";
+                // startRow parameter of "Sforce-Enable-PKChunking" header has to be a valid
+                // 15 or 18 char ID.
+                if (this.queryChunkStartRow != null 
+                    && (this.queryChunkStartRow.length() == 15 || this.queryChunkStartRow.length() == 18)) {
+                    startRowParam = "; startRow=" + this.queryChunkStartRow;
+                }
+                this.client.addHeader("Sforce-Enable-PKChunking", 
+                        "chunkSize=" + this.queryChunkSize
```

Comment:
Can we put line 151 and 152 into one for better readability? Even combining 151 and 512 to line 150 might be better if not too long.

Reply:
done.